python/pyiceberg/catalog/rest.py

```diff
@@ -489,6 +493,39 @@ def rename_table(self, from_identifier: Union[str, Identifier], to_identifier: U
 
         return self.load_table(to_identifier)
 
+    def _commit(self, *table_requests: CommitTableRequest) -> CommitTableResponse:
```

Comment:
Do we want to combine these? Why not have separate `commit_table` and `commit_transaction` methods? Then we don't need the check that only one request is supported. I also like that you could previously pass requirements and updates to a public method. Are you trying to restrict access to those in the "public" API for some reason?

Reply:
+1, given this is called afterwards in `Transaction`, it feels more nature to call it just `commit_table` so we an use `self._table.catalog.commit_table` instead of `self._table.catalog._commit`

Reply:
Yes, thinking of it, that makes sense. Also, the return type is different for a table and a transaction. Thanks!